libraries/crypto/src/sha256.rs

```diff
@@ -266,8 +266,8 @@ mod test {
         let hash = hex::decode("32beecb58a128af8248504600bd203dcc676adf41045300485655e6b8780a01d")
             .unwrap();
 
-        for i in 0..512 {
-            for j in i..512 {
+        for i in (0..512).step_by(16) {
+            for j in (i..512).step_by(16) {
```

Comment:
Why step 16 here? The idea of the test is to test slices of different sizes. I would use 3 and 5 (or 13 and 17 if we want bigger numbers)